### ISSUE TITLE
Upgrading from 4.2.3 to 4.2.4 is breaking MockMvc

### ISSUE THREAD

**Issue by edeandrea** [spring-projects/spring-security#5119]
I'm not sure the correct place to ask this - but moving from spring-security 4.2.3 to 4.2.4 it seems my tests that use MockMvc now break. 

If I do something like

`mockMvc.perform(MockMvcRequestBuilders.get("/somePath/Some text with space")`

Spring security 4.2.4 now barfs while spring security 4.2.3 works fine. This has to do with the new `StrictHttpFirewall` - its blocking the request because `MockMvc` is url-encoding and replacing the spaces with `%20`, which `StrictHttpFirewall` is now rejecting.

```
The request was rejected because the URL contained a potentially malicious String "%"
org.springframework.security.web.firewall.RequestRejectedException: The request was rejected because the URL contained a potentially malicious String "%"
```

What's weird though is that when running the application and hitting my endpoints via Postman, everything seems to be fine, so I'm not sure if this is a Spring security issue, Spring boot issue (we moved from 1.5.9 to 1.5.10), or a spring-test issue.

**Comment by rwinch** [comment]
Thanks for the report. This is a duplicate of https://jira.spring.io/browse/SPR-16453 which will be fixed when Spring Framework 4.3.15 is released. You can find a workaround for the issue in the "Work Around" section of the issue.

**Comment by edeandrea** [comment]
Thank you @rwinch
